src/Orchard.Web/Modules/Orchard.Fields/Drivers/BooleanFieldDriver.cs

```diff
@@ -35,7 +35,7 @@ public class BooleanFieldDriver : ContentFieldDriver<BooleanField> {
 
         protected override DriverResult Editor(ContentPart part, BooleanField field, dynamic shapeHelper) {
             // if the content item is new, assign the default value
-            if(!part.HasDraft() && !part.HasPublished()) {
+            if(!field.Value.HasValue) {
```

Comment:
The main problem here is that there is no way to know if the Boolean Field has been assigned a null value "explicitly"... since a null value is a null ...